ThirdPartyModules/Mirasvit/Rewards.php

```diff
@@ -0,0 +1,185 @@
+<?php
+/**
+ * Bolt magento2 plugin
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://opensource.org/licenses/osl-3.0.php
+ *
+ * @category   Bolt
+ * @package    Bolt_Boltpay
+ * @copyright  Copyright (c) 2017-2020 Bolt Financial, Inc (https://www.bolt.com)
+ * @license    http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+
+namespace Bolt\Boltpay\ThirdPartyModules\Mirasvit;
+
+use Bolt\Boltpay\Helper\Bugsnag;
+use Bolt\Boltpay\Helper\Shared\CurrencyUtils;
+use Bolt\Boltpay\Helper\Discount;
+use Magento\Store\Model\ScopeInterface;
+use Magento\Framework\App\Config\ScopeConfigInterface;
+
+class Rewards
+{
+    /**
+     * @var \Mirasvit\Rewards\Helper\Purchase
+     */
+    private $mirasvitRewardsPurchaseHelper;
+
+    /**
+     * @var Bugsnag
+     */
+    private $bugsnagHelper;
+
+    /**
+     * @var Discount
+     */
+    private $discountHelper;
+
+    /**
+     * @var ScopeConfigInterface
+     */
+    private $scopeConfigInterface;
+
+    /**
+     * Rewards constructor.
+     * @param Bugsnag $bugsnagHelper
+     * @param Discount $discountHelper
+     * @param ScopeInterface $scopeConfigInterface
+     */
+    public function __construct(
+        Bugsnag $bugsnagHelper,
+        Discount $discountHelper,
+        ScopeConfigInterface $scopeConfigInterface
+    )
+    {
+        $this->bugsnagHelper = $bugsnagHelper;
+        $this->discountHelper = $discountHelper;
+        $this->scopeConfigInterface = $scopeConfigInterface;
+    }
+
+    /**
+     * @param $immutableQuote
+     * @param $mirasvitRewardsPurchaseHelper
+     */
+    public function applyExternalDiscountData($immutableQuote, $mirasvitRewardsPurchaseHelper)
```

Comment:
It should be
```
public function applyExternalDiscountData($mirasvitRewardsPurchaseHelper, $immutableQuote)
```

Or there is a fatal error

```
Fatal Error: 'Uncaught Error: Call to undefined method Mirasvit\\Rewards\\Helper\\Purchase::getBoltParentQuoteId()
```

Reply:
@dumega Good catch. I've updated the code. Could you please re-check?